spec/controllers/lists_controller_spec.rb

```diff
@@ -44,6 +44,42 @@
     end
   end
 
+  describe 'searching', :sphinx do
+    login_user
+    let(:inc) { create(:entity_org) }
+
+    before do
+      setup_sphinx do
+        lists = [
+          create(:list, name: 'my interesting list'),
+          create(:list, name: 'someone else private list', access: Permissions::ACCESS_PRIVATE, creator_user_id: controller.current_user.id + 1),
+        ]
+        lists.each do |list|
+          ListEntity.find_or_create_by(list_id: list.id, entity_id: inc.id)
+        end
+      end
+    end
+
+    after(:all) do
+      teardown_sphinx {  }
+    end
+
+    it 'returns search results in HTML' do
+      get :index, params: { q: 'interesting' }
```

Comment:
@aepyornis some difficulties testing stuff that involves Sphinx. It seems I have to run `RAILS_ENV=test bin/rails ts:rebuild` every time I want to run a test involving Sphinx. It then spends ages on `Generating configuration to /littlesis/config/test.sphinx.conf`. Any suggestions?

```
ThinkingSphinx::SphinxError:
       unknown local index(es) 'list_core' in search request - SELECT * FROM `list_core`, `list_delta` WHERE MATCH('interesting') AND `is_deleted` = 0 AND `is_admin` = 0 AND `sphinx_deleted` = 0 LIMIT 0, 20; SHOW META
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/connection/client.rb:65:in `rescue in perform'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/connection/client.rb:61:in `perform'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/connection/client.rb:51:in `check_and_perform'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/connection/client.rb:36:in `query_all'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/search/batch_inquirer.rb:19:in `block in results'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/connection.rb:38:in `block in take'
     # /usr/local/bundle/gems/innertube-1.1.0/lib/innertube.rb:138:in `take'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/connection.rb:36:in `take'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/search/batch_inquirer.rb:18:in `results'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/middlewares/inquirer.rb:11:in `block in call'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/logger.rb:5:in `log'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/middlewares/inquirer.rb:10:in `call'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/middlewares/geographer.rb:13:in `call'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/middlewares/sphinxql.rb:16:in `call'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/middlewares/valid_options.rb:9:in `call'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/middlewares/stale_id_filter.rb:12:in `call'
     # /usr/local/bundle/gems/middleware-0.1.0/lib/middleware/runner.rb:31:in `call'
     # /usr/local/bundle/gems/middleware-0.1.0/lib/middleware/builder.rb:102:in `call'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/search.rb:95:in `populate'
     # /usr/local/bundle/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/search.rb:145:in `method_missing'
     # ./app/controllers/lists_controller.rb:59:in `index'
     # /usr/local/bundle/gems/rails-controller-testing-1.0.4/lib/rails/controller/testing/template_assertions.rb:61:in `process'
     # /usr/local/bundle/gems/devise-4.7.1/lib/devise/test/controller_helpers.rb:35:in `block in process'
     # /usr/local/bundle/gems/devise-4.7.1/lib/devise/test/controller_helpers.rb:102:in `catch'
     # /usr/local/bundle/gems/devise-4.7.1/lib/devise/test/controller_helpers.rb:102:in `_catch_warden'
     # /usr/local/bundle/gems/devise-4.7.1/lib/devise/test/controller_helpers.rb:35:in `process'
     # /usr/local/bundle/gems/rails-controller-testing-1.0.4/lib/rails/controller/testing/integration.rb:13:in `block (2 levels) in <module:Integration>'
     # ./spec/controllers/lists_controller_spec.rb:68:in `block (3 levels) in <main>'
     # /usr/local/bundle/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:55:in `load'
     # /usr/local/bundle/gems/bootsnap-1.4.6/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:55:in `load'
     # /usr/local/bundle/gems/spring-commands-rspec-1.0.4/lib/spring/commands/rspec.rb:18:in `call'
     # -e:1:in `<main>'
     # ------------------
     # --- Caused by: ---
     # Mysql2::Error:
     #   unknown local index(es) 'list_core' in search request
     #   /usr/local/bundle/gems/mysql2-0.5.3/lib/mysql2/client.rb:131:in `_query'
```

Reply:
Interesting. Are you running those tests outside docker?

While most tests run okay outside docker, the sphinx tests has never worked unless I run this _inside_ docker (`littlesis test spec/controllers/lists_controller_spec.rb  `)

But I've not seen that error before or at least not in some time.

Reply:
Thanks @aepyornis. I've tried both inside and outside the docker container (including installing Sphinx etc on my dev machine), and get similar results in both. Do you rebuild the indexes every time you want to run a test that involves Sphinx?